UnityProject/Assets/Scripts/Managers/GameManager.GameMode.cs

```diff
@@ -0,0 +1,98 @@
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+/// <summary>
+/// Represents what state the round is in: PreRound, Started or Ended
+/// </summary>
+public enum RoundState
+{
+	None,
+	PreRound,
+	Started,
+	Ended
+}
+
+/// <summary>
+/// Game mode related part of GameManager
+/// </summary>
+public partial class GameManager
+{
+	public List<GameMode> AllGameModes = new List<GameMode>();
+
+	/// <summary>
+	/// Is the current game mode being kept secret?
+	/// </summary>
+	public bool SecretGameMode = true;
+
+	/// <summary>
+	/// The state of the current round
+	/// </summary>
+	public RoundState CurrentRoundState;
+
+	/// <summary>
+	/// The current game mode
+	/// </summary>
+	private GameMode GameMode;
+
+	/// <summary>
+	/// Find all gamemode prefabs and add them to AllGameModes for selection
+	/// </summary>
+	private void RefreshAllGameModes()
+	{
+		// Only do this stuff on the server
+		if (CustomNetworkManager.Instance._isServer)
```

Comment:
For methods like this, where pretty much all the logic depends on a condition, imo it's a little bit cleaner to have it formatted as:

`if (!CustomNetworkManager.Instance._isServer) {
    return;
}

<logic>
`

Reply:
Fixed